spec/javascripts/app/models/notification_spec.js

```diff
@@ -0,0 +1,94 @@
+      "type": "reshared"
+    });
+  });
+
+  describe("constructor", function() {
+    it("calls parent constructor with the right parameters", function() {
+      spyOn(Backbone.Model.prototype, "constructor");
+      new app.models.Notification({attribute: "attribut"}, {option: "option"});
+      expect(Backbone.Model.prototype.constructor).toHaveBeenCalledWith(
+        {attribute: "attribut"},
+        {option: "option", parse: true}
+      );
+    });
+  });
+
+  describe("parse", function() {
+    it("correctly parses the object", function() {
+      var parsed = this.model.parse({
+        "reshared": {
+          "id": 45,
+          "target_type": "Post",
+          "target_id": 11,
+          "recipient_id": 1,
+          "unread": true,
+          "created_at": "2015-10-27T19:56:30.000Z",
+          "updated_at": "2015-10-27T19:56:30.000Z",
+          "note_html": "<html/>"
+        },
+        "type": "reshared"
+      });
+
+      expect(parsed).toEqual({
+        "type": "reshared",
+        "id": 45,
+        "target_type": "Post",
+        "target_id": 11,
+        "recipient_id": 1,
+        "unread": true,
+        "created_at": "2015-10-27T19:56:30.000Z",
+        "updated_at": "2015-10-27T19:56:30.000Z",
+        "note_html": "<html/>"
+      });
+    });
+  });
+
+  describe("getGUID", function() {
+    it("returns the correct ID", function() {
+      var target = this.model = new app.models.Notification({"reshared": {id: 16}, "type": "reshared"});
+      expect(target.getGUID()).toBe(16);
+    });
+  });
+
+  describe("setRead", function() {
+    it("calls setUnreadStatus with 'false'", function() {
+      spyOn(app.models.Notification.prototype, "setUnreadStatus");
+      new app.models.Notification({"reshared": {}, "type": "reshared"}).setRead();
+      expect(app.models.Notification.prototype.setUnreadStatus).toHaveBeenCalledWith(false);
+    });
+  });
+
+  describe("setUnread", function() {
+    it("calls setUnreadStatus with 'true'", function() {
+      spyOn(app.models.Notification.prototype, "setUnreadStatus");
+      new app.models.Notification({"reshared": {}, "type": "reshared"}).setUnread();
+      expect(app.models.Notification.prototype.setUnreadStatus).toHaveBeenCalledWith(true);
+    });
+  });
+
+  describe("setUnreadStatus", function() {
+    beforeEach(function() {
+      this.target = new app.models.Notification({"reshared": {id: 16}, "type": "reshared"});
+      spyOn(jQuery, "ajax").and.callThrough();
+      spyOn(app.models.Notification.prototype, "set").and.callThrough();
+    });
+
+    it("calls calls ajax with correct parameters and sets 'unread' attribute", function() {
+      this.target.setUnreadStatus(true);
+      jasmine.Ajax.requests.mostRecent().respondWith({status: 200, responseText: '{"guid": 16, "unread": true}'});
+      var call = jQuery.ajax.calls.mostRecent();
+
+      expect(jQuery.ajax).toHaveBeenCalled();
+      expect(call.args[0].url).toBe("/notifications/16");
```

Comment:
With `jasmine.Ajax.requests.mostRecent()` this would be
```javascript
expect(jasmine.Ajax.requests.mostRecent().url).toBe('/notifications/16')
```
or
```javascript
expect(call.url).toBe('/notifications/16')
```